contracts/discrepancies/nonce/InternalCallee.sol

```diff
@@ -15,14 +15,15 @@ contract InternalCallee {
     }
 
     function revertWithRevertReason() public returns (bool) {
+        if (calledTimes < 0) {++calledTimes;}
```

Comment:
Please add a comment to `calledTimes` so it's clear how it's intended to be used.
Right now it's not clear why `externalFunction()` and `revertWithRevertReason()` increment it but other functions don't

Reply:
The original warning was "Function state mutability can be restricted to pure", then if I made this function a pure function, the nonce of the caller won't be incremented when this method is called which is a case scenario in the UTs. That's why I added the line `if (calledTimes < 0) {++calledTimes}` which is an impossible condition but to make the compiler treat this as a non-view function. Hmm now that I look at this again this line maybe not needed to be here but we can fix the UT, or maybe the case scenario in UTs is the intended case. I'll reach out to Nikolay about this

Reply:
Reached out to Nikolay and he confirmed that we want to maintain the non-view status of the function so we can just simply add `++calledTimes;`. I pushed in the fix along with comments showing the intended purpose of the state mutation.